pkg/kubectl/cmd/convert.go

```diff
@@ -78,55 +99,17 @@ func NewCmdConvert(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *co
 
 	options.PrintFlags.AddFlags(cmd)
 
-	usage := "to need to get converted."
-	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
-	cmd.MarkFlagRequired("filename")
 	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, "to need to get converted.")
+	cmd.MarkFlagRequired("filename")
+
 	cmd.Flags().BoolVar(&options.local, "local", options.local, "If true, convert will NOT try to contact api-server but run locally.")
-	cmd.Flags().String("output-version", "", i18n.T("Output the formatted object with the given group version (for ex: 'extensions/v1beta1').)"))
+	cmd.Flags().StringVar(&options.OutputVersion, "output-version", "", i18n.T("Output the formatted object with the given group version (for ex: 'extensions/v1beta1').)"))
 	return cmd
 }
 
-// ConvertOptions have the data required to perform the convert operation
-type ConvertOptions struct {
-	PrintFlags *genericclioptions.PrintFlags
-	PrintObj   printers.ResourcePrinterFunc
-
-	resource.FilenameOptions
-
-	builder *resource.Builder
-	local   bool
-
-	genericclioptions.IOStreams
-	specifiedOutputVersion schema.GroupVersion
-}
-
-func NewConvertOptions(ioStreams genericclioptions.IOStreams) *ConvertOptions {
-	return &ConvertOptions{
-		PrintFlags: genericclioptions.NewPrintFlags("converted").WithTypeSetter(scheme.Scheme).WithDefaultOutput("yaml"),
-		local:      true,
-		IOStreams:  ioStreams,
-	}
-}
-
-// outputVersion returns the preferred output version for generic content (JSON, YAML, or templates)
-// defaultVersion is never mutated.  Nil simply allows clean passing in common usage from client.Config
-func outputVersion(cmd *cobra.Command) (schema.GroupVersion, error) {
-	outputVersionString := cmdutil.GetFlagString(cmd, "output-version")
-	if len(outputVersionString) == 0 {
-		return schema.GroupVersion{}, nil
-	}
-
-	return schema.ParseGroupVersion(outputVersionString)
-}
-
 // Complete collects information required to run Convert command from command line.
 func (o *ConvertOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) (err error) {
-	o.specifiedOutputVersion, err = outputVersion(cmd)
-	if err != nil {
-		return err
-	}
-
 	// build the builder
 	o.builder = f.NewBuilder().
```

Comment:
Move the builder construction entirely to `Run`, I don't think it's needed here.